c10/xpu/XPUFunctions.cpp

```diff
@@ -0,0 +1,285 @@
+  device_prop->pref_vec_width_char = raw_device.get_info<device::preferred_vector_width_char>();
+  device_prop->pref_vec_width_short = raw_device.get_info<device::preferred_vector_width_short>();
+  device_prop->pref_vec_width_int = raw_device.get_info<device::preferred_vector_width_int>();
+  device_prop->pref_vec_width_long = raw_device.get_info<device::preferred_vector_width_long>();
+  device_prop->pref_vec_width_float = raw_device.get_info<device::preferred_vector_width_float>();
+  device_prop->pref_vec_width_double = raw_device.get_info<device::preferred_vector_width_double>();
+  device_prop->pref_vec_width_half = raw_device.get_info<device::preferred_vector_width_half>();
+  device_prop->native_vec_width_char = raw_device.get_info<device::native_vector_width_char>();
+  device_prop->native_vec_width_short = raw_device.get_info<device::native_vector_width_short>();
+  device_prop->native_vec_width_int = raw_device.get_info<device::native_vector_width_int>();
+  device_prop->native_vec_width_long = raw_device.get_info<device::native_vector_width_long>();
+  device_prop->native_vec_width_float = raw_device.get_info<device::native_vector_width_float>();
+  device_prop->native_vec_width_double = raw_device.get_info<device::native_vector_width_double>();
+  device_prop->native_vec_width_half = raw_device.get_info<device::native_vector_width_half>();
+  // Intel extension
+  device_prop->gpu_eu_count = raw_device.has(sycl::aspect::ext_intel_gpu_eu_count)
+      ? raw_device.get_info<intel::info::device::gpu_eu_count>()
+      : 512;
+  device_prop->gpu_eu_count_per_subslice = raw_device.has(sycl::aspect::ext_intel_gpu_eu_count_per_subslice)
+      ? raw_device.get_info<intel::info::device::gpu_eu_count_per_subslice>()
+      : 8;
+  device_prop->gpu_eu_simd_width = raw_device.has(sycl::aspect::ext_intel_gpu_eu_simd_width)
+      ? raw_device.get_info<intel::info::device::gpu_eu_simd_width>()
+      : 8;
+  device_prop->gpu_hw_threads_per_eu = raw_device.has(sycl::aspect::ext_intel_gpu_hw_threads_per_eu)
+      ? raw_device.get_info<intel::info::device::gpu_hw_threads_per_eu>()
+      : 8;
+  // clang-format on
+  return;
+}
+
+/*
+ * Note [Runtime in Multiprocessing]
+ *
+ * We have known the limitation of fork support in SYCL runtime. If we call
+ * runtime APIs in parent process, then fork a child process, there will be an
+ * error in runtime if submit any kernel in parent process or child process.
+ *
+ * In general, SYCL runtime initialization must be called after fork, not
+ * before. So we have to call runtime APIs using another fork, pipe the result
+ * back to the parent, and then fork the actual child process.
+ *
+ * We have to fork another child process first. Then query device count using
+ * SYCL runtime APIs. Finally pipe the result to parent process. Now we can
+ * check if XPU device is available and fork the actual child process to do the
+ * calculation.
+ */
+
+bool prefetchDeviceCount(int* device_count) {
```

Comment:
Can you elaborate more on the use case of this function?

Reply:
this function is used for `prefetch_device_count`. And I give some comments for `prefetch_device_count`